classes/class-wc-connect-service-settings-store.php

```diff
@@ -43,54 +43,79 @@ public function get_shared_settings() {
 		}
 
 		/**
-		 * Given a service's id and optional instance, returns the settings for that
-		 * service or an empty array
-		 *
-		 * @param string $service_id
-		 * @param integer $service_instance
+		 * Returns the service type and id for each enabled WooCommerce Connect service
 		 *
-		 * @return object|array
-		 */
-		public function get_service_settings( $service_id, $service_instance = false ) {
-			return get_option( $this->get_service_settings_key( $service_id, $service_instance ), array() );
-		}
-
-		/**
-		 * Returns a nonce action based on the service id and optional instance,
-		 * suitable for use in generating or validating nonces
+		 * Shipping services also include instance_id and shipping zone id
 		 *
-		 * @param string $service_id
-		 * @param integer $service_instance
+		 * Note that at this time, only shipping services exist, but this method will
+		 * return other services in the future
 		 *
-		 * @return string
+		 * @return array
 		 */
-		protected function get_wc_api_callback_nonce_action( $service_id, $service_instance = false ) {
-			if ( ! $service_instance ) {
-				return 'wc_connect_' . $service_id;
+		public function get_enabled_services() {
+
+			$enabled_services = array();
+
+			$shipping_services = $this->service_schemas_store->get_all_service_ids_of_type( 'shipping' );
+			if ( empty( $shipping_services ) ) {
+				return $enabled_services;
+			}
+
+			// Note: We use esc_sql here instead of prepare because we are using WHERE IN
+			// https://codex.wordpress.org/Function_Reference/esc_sql
+
+			$escaped_list = '';
+			foreach ( $shipping_services as $shipping_service ) {
+				if ( ! empty( $escaped_list ) ) {
+					$escaped_list .= ',';
+				}
+				$escaped_list .= "'" . esc_sql( $shipping_service ) . "'";
 			}
-			return 'wc_connect_' . $service_id . '_' . $service_instance;
+
+			global $wpdb;
+			$methods = $wpdb->get_results(
+				"SELECT * FROM {$wpdb->prefix}woocommerce_shipping_zone_methods " .
+				"INNER JOIN {$wpdb->prefix}woocommerce_shipping_zones " .
+				"ON {$wpdb->prefix}woocommerce_shipping_zone_methods.zone_id = {$wpdb->prefix}woocommerce_shipping_zones.zone_id " .
```

Comment:
The "Rest of World" default zone does not have an entry in `woocommerce_shipping_zones`. This query therefore doesn't get back any results for shipping methods added to "Rest of World".